teammapper-frontend/mmp/src/map/handlers/history.ts

```diff
@@ -26,6 +27,26 @@ export default class History {
         this.snapshots = []
     }
 
+    /**
+     * As snapshots are saved by number (1, 2, 3 etc) and not by node ID, this switches out the number with the affected node ID.
+     * This helps the server understand which node was modified.
+     * @param {MapDiff} snapshotDiff
+     */
+    private switchDiffKeys(snapshotDiff: MapDiff): void {
+        ['added', 'deleted', 'updated'].forEach((key: keyof MapDiff) => {
+            const diffSection = snapshotDiff[key];
+            if (diffSection && typeof diffSection === 'object') {
+                Object.entries(diffSection).forEach(([index, value]: [string, Partial<ExportNodeProperties> | undefined]) => {
+                    const nodeId = value?.id ?? this.snapshots[this.index][Number(index)]?.id;
+                    if (nodeId) {
+                        delete diffSection[Number(index)];
```

Comment:
I would propose a more functional way of doing this. Not modifying an existing data structure in place, but to create a new value without the to deleted value. E.g. by using `filter`.